tests/conftest.py

```diff
@@ -0,0 +1,156 @@
+import json
+import os
+import random
+import shutil
+import string
+import subprocess
+import time
+import urllib.request
+
+import pytest
+from ephemeral_port_reserve import reserve
+
+from docker_push_latest_if_changed import _inspect_image
+
+
+DOCKER_REGISTRY_IMAGE_NAME = 'registry'
+DEFAULT_TESTING_IMAGES_PATH = os.path.abspath(
+    os.path.join('testing', 'dockerfiles')
+)
+
+
+@pytest.fixture
+def fake_docker_registry():
+    subprocess.check_call(('docker', 'pull', DOCKER_REGISTRY_IMAGE_NAME))
```

Comment:
useless constant: only used once, inlining would actually be much more readable

also you don't need to `pull`, the `run` command will pull if it needs to.

Also you should probably select a specific major version of the registry, so `registry:2` for instance

Reply:
Though unlikely to change, the image name for registry on the registry is still an external variable and setting it as its own constant highlights that. I'd like to keep it, but if you feel strongly against it I'll inline it.

Will remove the pull and add the version tag though.